include/mqtt/utf8encoded_strings.hpp

```diff
@@ -13,18 +13,114 @@ namespace mqtt {
 
 namespace utf8string {
 
+enum struct validation
+{
+    /**
+     * @brief UTF-8 string is well_formed.
+     * See http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718016
+     * 1.5.3 UTF-8 encoded strings
+     */
+    well_formed = 0,
+
+    /**
+     * @brief UTF-8 string is ill_formed or contains null character.
+     * See http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718016
+     * 1.5.3 UTF-8 encoded strings
+     */
+    ill_formed,
+
+    /**
+     * @brief UTF-8 string is well_formed and contains control character and non-character.
+     * See http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718016
+     * 1.5.3 UTF-8 encoded strings
+     */
+    well_formed_with_non_charactor,
+};
+
 inline bool
 is_valid_length(std::string const& str) {
     return str.size() <= 0xffff;
 }
 
-inline bool
-is_valid_contents(std::string const& /*str*/) {
-    // not implemented yet
-    // See 1.5.3 UTF-8 encoded strings
-    return true;
+inline validation
+validate_contents(std::string const& str) {
+    // This code is based on https://www.cl.cam.ac.uk/~mgk25/ucs/utf8_check.c
+    auto result = validation::well_formed;
+#if !defined(MQTT_USE_STR_CK)
```

Comment:
I think that `CK` is short form of `CHECK`. I considered `CK`, `CHK`, and `CHECK`. `CHECK` is better here.
Could you replace it?